test/e2e/lib/gutenberg/gutenberg-editor-component.js

```diff
@@ -71,10 +74,14 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
 	async publish( { visit = false, closePanel = true } = {} ) {
 		await driverHelper.clickWhenClickable( this.driver, this.prePublishButtonSelector );
 		await driverHelper.clickWhenClickable( this.driver, this.publishButtonSelector );
-		// Let's give publishing request enough time to finish. Sometimes it takes
-		// way more than the default 20 seconds, and the cost of waiting a bit
-		// longer is definitely lower than the cost of repeating the whole spec.
-		await driverHelper.waitTillNotPresent( this.driver, this.publishingSpinnerSelector, 60000 );
+
+		// When publishing request completes, the close button appears.
+		// We use the existence of the close button to determine that the publishing request is completed
+		// before moving on to the next step.
+		await driverHelper.waitUntilLocatedAndVisible(
+			this.driver,
+			this.closePublishPanelButtonSelector
+		);
```

Comment:
FYI this is making the [publishing flow fail](https://teamcity.a8c.com/viewLog.html?buildId=6136691&buildTypeId=calypso_Gutenberg&tab=buildResultsDiv&branch_calypso_WPComPlugins=fix%2Fe2e-publish-post) for the payment block because of https://github.com/Automattic/wp-calypso/issues/50302. I'm addressing this issue in https://github.com/Automattic/wp-calypso/pull/52089.

Reply:
https://github.com/Automattic/wp-calypso/pull/52089/commits/3fd6f256e790d810846c0ac684f41802be2838ed